python/lance/util/duckdb_ext.py

```diff
@@ -0,0 +1,68 @@
+import duckdb
+import os
+import pathlib
+import platform
+import urllib.request
+import zipfile
+
+
+def install_duckdb_extension(version='latest', unsigned=True):
+    """
+    Install the lance duckdb extension
+
+    Parameters
+    ----------
+    version: str, default 'latest'
+        The version of the extension to install
+    unsigned: bool, default True
+        Whether to turn on allow_unsigned_extensions in duckdb
+    """
+    if version == 'latest':
+        version = _get_latest_version('lance', 'lance_duckdb')
+    uri = _get_uri(version)
+    local_path = _download_and_unzip('lance', 'lance_duckdb', uri)
+    con = duckdb.connect(config={"allow_unsigned_extensions": unsigned})
+    con.install_extension(local_path, force_install=True)
+
+
+def _get_uri(version):
+    uname = platform.uname()
+    arch = uname.machine  # arm64, x86_64
+    system = uname.system
+    system = "osx" if system.lower() == "darwin" else system
+    device = _get_device()
+    zip_name = f'lance.duckdb_extension.{system}.{arch}.{device}.zip'
+    uri_root = 'https://eto-public.s3.us-west-2.amazonaws.com/'
+    uri = os.path.join(uri_root, 'artifacts', 'lance', 'lance_duckdb',
+                       version, zip_name)
+    return uri
+
+
+def _get_device():
+    import torch
+    if torch.cuda.is_available():
+        return f"cu{torch.version.cuda.replace('.', '')}"
+    else:
+        return 'cpu'
+
+
+def _get_latest_version(org='lance', ext='lance_duckdb'):
+    uri_root = 'https://eto-public.s3.us-west-2.amazonaws.com/'
+    uri = os.path.join(uri_root, 'artifacts', org, ext, 'latest')
```

Comment:
should we check duckdb version and torch version?

i.e., can duckdb `0.5.0` loads the extension built against `0.5.1`

Reply:
good question i can try it first. If it's at least binary compatible across minor versions then we can build against 0.X.0 and just check the version

Reply:
ok so unfortunately the answer is no:

```python
In [5]: con.load_extension("lance")
---------------------------------------------------------------------------
InvalidInputException                     Traceback (most recent call last)
Cell In [5], line 1
----> 1 con.load_extension("lance")

InvalidInputException: Invalid Input Error: Extension "/Users/changshe/.duckdb/extensions/109f932c4/osx_arm64/lance.duckdb_extension" version (v0.5.1) does not match DuckDB version (0.5.0)
```